packages/flutter_tools/lib/src/web/devfs_proxy.dart

```diff
@@ -0,0 +1,107 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:yaml/yaml.dart';
+import '../base/logger.dart';
+import '../globals.dart' as globals;
+
+abstract class ProxyRule {
```

Comment:
Can this class be `sealed`? Then we'll have support for exhaustiveness checking.